quickstarts/EAP7-1940-add-http-custom-mechanism-to-EAP-quickstarts.adoc

```diff
@@ -0,0 +1,118 @@
+= Add http-custom-mechanism to the EAP quickstarts
+:author:            Farah Juma
+:email:             fjuma@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+
+== Overview
+
+The http-custom-mechanism quickstart is currently a community-only quickstart because it's not part of the EAP quickstarts repo.
+Nothing else demonstrates how to configure this so it should be added to the EAP quickstarts as well.
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.redhat.com/browse/EAP7-1940[EAP7-1940]
+* https://issues.redhat.com/browse/WFLY-16471[WFLY-16471]
+
+=== Related Issues
+
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+=== Testing By
+// Put an x in the relevant field to indicate if testing will be done by Engineering or QE. 
+// Discuss with QE during the Kickoff state to decide this
+* [ ] Engineering
+
+* [ ] QE
+
+=== Affected Projects or Components
+
+=== Other Interested Projects
+
+=== Relevant Installation Types
```

Comment:
According to https://github.com/wildfly/quickstart/blob/main/http-custom-mechanism/README.adoc
- The QS is not intended for managed domain (no such instructions).
- The QS is not supported on OpenShift. There is nothing about such exception in this analysis.

Reply:
@fjuma can you please add the following clarification:

- The QS is not intended for managed domain
- The QS is not supported on OpenShift.

Please note that I think we could make it openshift compatible, but we can do that "later", without the need of an EAP7, let's focus here on adding the current upstream only quickstart to EAP.

Reply:
Updated.

Reply:
Is there a technical reason why the QS is not supported on OpenShift? (I.e. it's not possible to use the demonstrated functionality on OpenShift.) If not, the implementation should be performed in scope of the given EAP7 RFE issue.

Reply:
This could likely be made OpenShift compatible but would require some work.

Reply:
@fjuma we can do it later using a WFLY+JBEAP if you think it's important, and @kstekovi then updates the testsuite,  we don't need to hold this EAP7 for such enhancing.

Reply:
@emmartins we will need an EAP7 issue to add support for OpenShift. It is a new supported platform for the QS which has an impact on QE.

Moreover, after reducing the scope, has the size changed? I see "S", is it still valid? AFAIK QE work should be lesser without OpenShift support (the test is a bit more complicated). cc @darranl

Reply:
Having the OpenShift part implement we would also demonstrate how to implement and use a custom module on OpenShift. I'm not aware of any other QS demonstrating such feature.

Reply:
@istraka we never reduced scope, it was never considered to add Openshift support. Regarding the need for an EAP7 to add OpenShift support, IMHO it's not worth to go through the process and have a feature team to enhance an existent quickstart, a WFLY/JBEAP linked to a QE JIRA should be enough for this.

Reply:
> Having the OpenShift part implement we would also demonstrate how to implement and use a custom module on OpenShift. I'm not aware of any other QS demonstrating such feature.

Wrt custom modules there is already [EAP7-1981](https://issues.redhat.com/browse/EAP7-1981).

Reply:
I hear you, however, adding new supported platform for a feature, functionality or a deliverable has impact on QE since we commit to support such deliverable according to EAP lifecycle support policies. It also needs a documentation (at least release notes) and so on. We won't be able to add support for such enhancement without EAP7 issue. If we want to add OpenShift support eventually, can we consider to add it now?